client/src/util/helpers.ts

```diff
@@ -82,6 +83,27 @@ export const longDateOptions = { year: "numeric", month: "short", day: "numeric"
 export const mediumDateOptions = { year: "numeric", month: "long" };
 export const shortDateOptions = { month: "short" };
 
+/**
+ * Delay the action of a certian function by a set amount of time.
+ *
+ * Originally copied from:
+ * https://gist.github.com/gragland/4e3d9b1c934a18dc76f585350f97e321#gistcomment-3073492
+ */
+const debounce = (delay: number, fn: any) => {
```

Comment:
btw @toolness I tried using the react hook version of this function that you implemented in tenants2, but given that I wanted to use it inside the `useWindowSize` hook function below, it ended up causing some rule-of-hooks type errors. I reverted back to this implementation.